interpreter/src/main/scala/pl/touk/nussknacker/engine/testmode/TestComponentsHolder.scala

```diff
@@ -0,0 +1,34 @@
+package pl.touk.nussknacker.engine.testmode
+
+import pl.touk.nussknacker.engine.api.component.Component
+import pl.touk.nussknacker.engine.api.process.WithCategories
+
+import java.util.UUID
+import scala.reflect.ClassTag
+
+case class TestComponentHolder(runId: TestRunId) extends Serializable {
+
+  def components[T <: Component: ClassTag]: Map[String, WithCategories[T]] = TestComponentsHolder.resultsForId[T](runId)
+
+  def clean(): Unit = TestComponentsHolder.cleanResult(runId)
+}
+
+object TestComponentsHolder {
+
+  private var results = Map[TestRunId, Map[String, WithCategories[Component]]]()
```

Comment:
I wonder, these are not results but components?

Reply:
Let's remove WithCategories from here, it will be deprecated in some time, let's either have ComponentDefinition or Component

Reply:
changed to ComponentDefinition